OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/ChangesLogRepositoryTests.cs

```diff
@@ -0,0 +1,253 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using NUnit.Framework;
+using OutOfSchool.Services;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.Tests.Common.TestDataGenerators;
+using OutOfSchool.WebApi.Services;
+
+namespace OutOfSchool.WebApi.Tests.Services.Database
+{
+    [TestFixture]
+    public class ChangesLogRepositoryTests
+    {
+        private DbContextOptions<OutOfSchoolDbContext> dbContextOptions;
+        private ValueProjector valueProjector = new ValueProjector();
+
+        private Provider provider;
+        private User user;
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            provider = ProvidersGenerator.Generate();
+            provider.LegalAddress.Id = 6;
+            provider.LegalAddressId = 6;
+            user = UserGenerator.Generate();
+
+            dbContextOptions = new DbContextOptionsBuilder<OutOfSchoolDbContext>()
+                .UseInMemoryDatabase(databaseName: "OutOfSchoolTestDB")
+                .EnableSensitiveDataLogging()
+                .Options;
+
+            await Seed();
+        }
+
+        #region AddChangesLogToDbContext
+        [Test]
+        public async Task AddChangesLogToDbContext_WhenEntityIsModified_AddsLogToDbContext()
+        {
+            // Arrange
+            using var context = GetContext();
+            var changesLogRepository = GetChangesLogRepository(context);
+            var trackedProperties = new[] { "FullTitle", "EdrpouIpn", "Director", "LegalAddress" };
+            var provider = await context.Providers.FirstOrDefaultAsync();
+
+            var oldFullTitle = provider.FullTitle;
+            var oldDirector = provider.Director;
+
+            // Act
+            provider.FullTitle += "new";
+            provider.Director += "new";
+
+            var newFullTitle = provider.FullTitle;
+            var newDirector = provider.Director;
+
+            var added = changesLogRepository.AddChangesLogToDbContext(provider, user.Id, trackedProperties, valueProjector.ProjectValue);
+
+            // Assert
+            var fullTitleChanges = context.ChangesLog.Local
```

Comment:
In some places `nameof(Provider)`, `nameof(Provider.FullTitle)` and the same in `trackedProperties`

Reply:
I agree. But, in my opinion, using string literals in tests is more readable. In any case, it does not have a negative impact.

Reply:
The main problem literals in such form is that they must be changed in all places every time your property name is changed. You've already used `nameof()` in `ArgumentNullException()`. You could use literal string there as well, why couldn't you? :)